src/components/SideNav.js

```diff
@@ -55,6 +63,23 @@ export default function SideNav({ setKeyword, children }) {
     getAppwriteUser();
   }, []);
 
+  useDebounce(
+    () => {
+      setDebouncedSearchKeyword(searchKeyword);
+      setKeyword(searchKeyword);
+    },
+    200,
+    [searchKeyword],
+  );
+
+  useEffect(() => {
+    if (!debouncedSearchKeyword) {
+      router.push(pathName);
+    } else {
+      router.push(`?keyword=${debouncedSearchKeyword}`);
+    }
+  }, [debouncedSearchKeyword, router, pathName]);
```

Comment:
Is `pathName` really needed in dependency array elements

Reply:
it is used at line 77, though I don't see any impact keeping/removing it from the dependency array